cni.go

```diff
@@ -139,15 +139,23 @@ func (c *libcni) Networks() []*Network {
 	return append([]*Network{}, c.networks...)
 }
 
-// Setup setups the network in the namespace
-func (c *libcni) Setup(ctx context.Context, id string, path string, opts ...NamespaceOpts) (*CNIResult, error) {
+// Setup setups the network in the namespace and returns a Result
+func (c *libcni) Setup(ctx context.Context, id string, path string, opts ...NamespaceOpts) (*Result, error) {
 	if err := c.Status(); err != nil {
 		return nil, err
 	}
 	ns, err := newNamespace(id, path, opts...)
 	if err != nil {
 		return nil, err
 	}
+	result, err := c.attachNetworks(ctx, ns)
+	if err != nil {
+		return nil, err
+	}
+	return c.createResult(result)
+}
+
+func (c *libcni) attachNetworks(ctx context.Context, ns *Namespace) ([]*current.Result, error) {
 	var results []*current.Result
 	for _, network := range c.Networks() {
```

Comment:
Is it expected that `Setup()` acts on a _copy_ of `libcni.networks[]`? If I see correctly, `libcni.Networks()` creates a new slice; https://github.com/containerd/go-cni/blob/8fbf3637eb5f67bb16eaacd4ec23a82e4d33b3ec/cni.go#L139

(added in https://github.com/containerd/go-cni/commit/e7a8c33b6ae91e33bf6a4a43ab511f9ae000b7b6), but the function mentions "results should not be modified"; https://github.com/containerd/go-cni/blob/8fbf3637eb5f67bb16eaacd4ec23a82e4d33b3ec/cni.go#L134-L140

Wondering if this should act on `c.networks` directly